clang/test/CIR/CodeGen/complex-cast.c

```diff
@@ -0,0 +1,216 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -clangir-disable-passes -emit-cir -o %t.cir %s
+// RUN: FileCheck --input-file=%t.cir --check-prefixes=CLANG,CIRGEN,CHECK %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -x c++ -fclangir -clangir-disable-passes -emit-cir -o %t.cir %s
+// RUN: FileCheck --input-file=%t.cir --check-prefixes=CPPLANG,CIRGEN,CHECK %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir -o %t.cir %s
+// RUN: FileCheck --input-file=%t.cir --check-prefixes=CLANG,CIR,CHECK %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -x c++ -fclangir -emit-cir -o %t.cir %s
+// RUN: FileCheck --input-file=%t.cir --check-prefixes=CPPLANG,CIR,CHECK %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm -o %t.ll %s
+// RUN: FileCheck --input-file=%t.ll --check-prefixes=CLANG,LLVM,CHECK %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -x c++ -fclangir -emit-llvm -o %t.ll %s
+// RUN: FileCheck --input-file=%t.ll --check-prefixes=CPPLANG,LLVM,CHECK %s
```

Comment:
I think we can reduce the number of compiler invocations a bit, is there anything specific you want to check for `-clangir-disable-passes`? I'd say 2 compiler invocation should be enough, one for CIR and one for LLVM (specially if C and C++ should emit the same code for the part that matters).

Reply:
There are just too many invocations and multiple check prefixes in the same line, not something we usually do, unless necessary, which doesn't seem to be the case. Please reduce this to 3 compiler invocations:
- Remove the c++ variants.
- Instead of `-clangir-disable-passes` either (a) use `--mlir-print-ir-{before,after}=pass-name` (see `static.cpp` for examples) or (b) add a comment to explain what it means for the purposes of the test.
- 3 CHECKs, no more than one per FileCheck invocation. They should be LLVM, CIR_BEFORE, CIR_AFTER (or some name like these)
- Checking the function name among different prefixes is fine, no need to optimize for that (like you did with `CLANG`).

Reply:
Updated.